Documentation/ceph-cluster-crd.md

```diff
@@ -1489,3 +1490,42 @@ data:
 
 Note: if you are using self-signed certificates (not known/approved by a proper CA) you must pass `VAULT_SKIP_VERIFY: true`.
 Communications will remain encrypted but the validity of the certificate will not be verified.
+
+##### Ceph CSI configuration
+
+Ceph-CSI is capable of encrypting requested PVs.
+Rook can configure Ceph-CSI to turn on encryption with Key Management System support.
+For this, you need to add new parameter in your CephCluster CR spec:
+
+```yaml
+security:
+  kms:
+  ...
+  ...
+  csiConnectionDetailsCMName: "rook-ceph-csi-vault"
+```
+
+The config map will look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rook-ceph-csi-vault
+  namespace: rook-ceph
+data:
+  csiConfig: |-
+    {
+      "vault-test": {
+        "encryptionKMSType": "vault",
+        "vaultAddress": "http://vault.default.svc.cluster.local:8200",
+        "vaultAuthPath": "/v1/auth/kubernetes/login",
+        "vaultRole": "csi-kubernetes",
```

Comment:
the role is not used, that is only for the `vault` type of KMS in Ceph-CSI, and refers to the Kubernetes ServiceAccount.